vars/checkDockerImage.groovy

```diff
@@ -0,0 +1,41 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+  Checks if the given Docker image exists.
+
+  whenTrue(checkDockerImage(image: 'hello-world:latest')) {
+    ...
+  }
+*/
+def call(Map args = [:]) {
+  def image = args.containsKey('image') ? args.image : error('checkDockerImage: image parameter is required')
+
+  def redirectStdout = isUnix() ? '2>/dev/null' : '2>NUL'
+  if (cmd(returnStatus: true, script: "docker images -q ${image}  ${redirectStdout}") == 0) {
```

Comment:
What happen if docker is not installed? For instance, in CI `macOS` workers is required to prepare the docker environment context, otherwise docker won't be ready, IIRC.

You could potentially use `isInstalled(tool: 'docker', flag: '--version')` for that particular validation

Reply:
Cool, I think it's important to check the tools. Thanks for the suggestion, will submit a commit with it ASAP

Reply:
Added in dacf11682ab6a446ee5e89a9d860fab4c82861ea